src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -2014,4 +2138,594 @@ void CorProfiler::GetAssemblyAndSymbolsBytes(BYTE** pAssemblyArray, int* assembl
+  ULONG callTargetReturnIndex = static_cast<ULONG>(ULONG_MAX);
+  ULONG returnValueIndex = static_cast<ULONG>(ULONG_MAX);
+  mdToken callTargetStateToken = mdTokenNil;
+  mdToken exceptionToken = mdTokenNil;
+  mdToken callTargetReturnToken = mdTokenNil;
+  ILInstr* firstInstruction;
+  callTargetTokens->ModifyLocalSigAndInitialize(&reWriterWrapper, caller, 
+      &callTargetStateIndex, &exceptionIndex, 
+      &callTargetReturnIndex, &returnValueIndex, 
+      &callTargetStateToken,
+      &exceptionToken, &callTargetReturnToken, &firstInstruction);
+
+  // ***
+  // BEGIN METHOD PART
+  // ***
+
+  // *** Load instance into the stack (if not static)
+  if (isStatic) {
+    if (caller->type.valueType) {
+      // Static methods in a ValueType can't be instrumented. 
+      // In the future this can be supported by adding a local for the valuetype and initialize it to the default value.
+      // After the signature modification we need to emit the following IL to initialize and load into the stack.
+      //    ldloca.s [localIndex]
+      //    initobj [valueType]
+      //    ldloc.s [localIndex]
+      Warn("*** CallTarget_RewriterCallback(): Static methods in a ValueType cannot be instrumented. ");
+      return E_FAIL;
+    }
+    reWriterWrapper.LoadNull();
+  } else {
+    reWriterWrapper.LoadArgument(0);
+    if (caller->type.valueType) {
+      if (caller->type.type_spec != mdTypeSpecNil) {
+        reWriterWrapper.LoadObj(caller->type.type_spec);
+      } 
+      if (caller->type.name.find("`1"_W) == std::string::npos) {
+        reWriterWrapper.LoadObj(caller->type.id);
+      }
+    }
+  }
+
+  // *** Load the method arguments to the stack
+  unsigned elementType;
+  if (numArgs <= 6) {
+    // Load the arguments directly (FastPath)
+    for (int i = 0; i < numArgs; i++) {
+      reWriterWrapper.LoadArgument(i + (isStatic ? 0 : 1));
+      auto argTypeFlags = methodArguments[i].GetTypeFlags(elementType);
+      if (argTypeFlags & TypeFlagByRef) {
+        Warn(
+            "*** CallTarget_RewriterCallback(): Methods with ref parameters "
+            "cannot be instrumented. ");
+        return E_FAIL;
+      }
+    }
+  } else {
+    // Load the arguments inside an object array (SlowPath)
+    reWriterWrapper.CreateArray(callTargetTokens->GetObjectTypeRef(), numArgs);
+    for (int i = 0; i < numArgs; i++) {
+      reWriterWrapper.BeginLoadValueIntoArray(i);
+      reWriterWrapper.LoadArgument(i + (isStatic ? 0 : 1));
+      auto argTypeFlags = methodArguments[i].GetTypeFlags(elementType);
+      if (argTypeFlags & TypeFlagByRef) {
+        Warn(
+            "*** CallTarget_RewriterCallback(): Methods with ref parameters "
+            "cannot be instrumented. ");
+        return E_FAIL;
+      }
+      if (argTypeFlags & TypeFlagBoxedType) {
+        auto tok = methodArguments[i].GetTypeTok(
+            metaEmit, callTargetTokens->GetCorLibAssemblyRef());
+        if (tok == mdTokenNil) {
+          return S_OK;
```

Comment:
How come we return `S_OK` here instead of `E_FAIL`?

Reply:
you're right should be `E_FAIL`

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/9e7bcb3955022de47db4e80e29c33db9f64b69c2